core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -123,7 +123,7 @@ private[spark] class TaskSetManager(
   // state until all tasks have finished running; we keep TaskSetManagers that are in the zombie
   // state in order to continue to track and account for the running tasks.
   // TODO: We should kill any running task attempts when the task set manager becomes a zombie.
-  private[scheduler] var isZombie = false
+  @transient private[scheduler] var isZombie = false
```

Comment:
This is transient because ... if it's true then that fact shouldn't survive serialization?
Just wondering if volatile was what was meant.

Reply:
previously, only the task completion/fail event can mutate it, which is in the task-result-getter thread.

Now, DAGScheduler can also mutate it via `TaskSchedulerImpl.submitTasks`, which is in the dag-scheduler-event-loop thread.

Reply:
damn, mistyped it...